cli/server/server.go

```diff
@@ -174,6 +174,9 @@ func dumpDB(ctx *cli.Context) error {
 		if err != nil {
 			return cli.NewExitError(fmt.Errorf("failed to get block %d: %s", i, err), 1)
 		}
+		buf := io.NewBufBinWriter()
+		b.EncodeBinary(buf.BinWriter)
```

Comment:
Double encoding? Please, don't. Either use `GetVarSize()`, or better (as `GetVarSize()` also does encoding) just write the buffer you have now below.